OMN Token Bridge

Mi-Lan opened the pull request:
How to Test:
- Since Omen Token is not deployed i used GEN erc-20 in the background...Get some gen
Deposit:
- Go into Your connection and select deposit
- Then approve Omen Token bridge
- Desposit some amount 
Withdraw:
- Go into your connection modal
- Select withdraw
- Enter some amount 
- Since claim option is not scope of this pr only way to confirm if transaction is successful is to go into omni bridge ui and confirm heres link https://omni.xdaichain.com/history

Mi-Lan committed:
restructured your connection to include separete claim section

Mi-Lan committed:
included omen icon for balances which will be changed

Mi-Lan committed:
implemented radio button and reorganized deposit section

Mi-Lan committed:
correct value shown for exchange balance and input field

Mi-Lan committed:
implemented deposit into current xdai service structure

Mi-Lan committed:
added dummy gen address

Mi-Lan committed:
getting deposit function to pass tough mofo

Mi-Lan committed:
debug code

Mi-Lan committed:
fighting battle with withdraw relay error payment fee not found

Mi-Lan committed:
cracked this motherfucker like an egg took me long time but nevertheless

Mi-Lan committed:
cleaned up code and implemented withdraw

Mi-Lan committed:
merged conflicts

Mi-Lan committed:
cleaned up code from debugging some more

Mi-Lan committed:
deleted unused code

Mi-Lan committed:
styled fields,added omen balance fetch in connectedBalance added icons

Mi-Lan committed:
Merge branch 'master' of https://github.com/protofire/omen-exchange into feature/1816-omn-bridge

Mi-Lan committed:
set allowance added with logic implemented

Mi-Lan committed:
implemented set allowance for deposit with working states

Mi-Lan committed:
started work on withdrawa and renamed vars

Mi-Lan committed:
hooked up omen mainnet balance to display and tried to hook up xDai with useCollateral but had trouble using static value for now

Mi-Lan committed:
cleaned up code and hooked up withdraw balance to work along with allowance

Mi-Lan committed:
Merge branch 'master' of https://github.com/protofire/omen-exchange into feature/1816-omn-bridge

Mi-Lan committed:
renamed bunch of stuff and cleaned up code

Mi-Lan committed:
fixed alert icon

Mi-Lan committed:
changed modal transaction to encompass icons and fixed all icons

Mi-Lan committed:
fixed errors

Mi-Lan committed:
styled allowance and added hover styling along with bunch of fixes

pimato committed:
Merge branch 'master' into feature/1816-omn-bridge

pimato reviewed (changes requested):
@Mi-Lan I am getting a compile warning:
```Compiled with warnings.

./src/hooks/useCollateralBalance.tsx
  Line 35:5:  Unexpected console statement  no-console```

pimato commented:
Margin between the Inputfield and the required amount info message should be 16px and not 12px:
<img width="439" alt="Bildschirmfoto 2021-04-21 um 09 59 44" src="https://user-images.githubusercontent.com/5337809/115518366-5cbbde00-a288-11eb-94a9-b26f3cd16133.png">

pimato commented:
After closing the confirmed transaction modal I went to the `Your Connection - Modal` and it seems like the balance is not getting updated after we did a successful deposit:
<img width="440" alt="Bildschirmfoto 2021-04-21 um 11 11 22" src="https://user-images.githubusercontent.com/5337809/115528568-73ffc900-a292-11eb-8fd5-45f1a531e90d.png">

pimato commented:
Once the approval has been set we should not remove it but rather show it like we do on Market creation process:
<img width="617" alt="Bildschirmfoto 2021-04-21 um 11 23 29" src="https://user-images.githubusercontent.com/5337809/115530756-5af81780-a294-11eb-89b2-9c86c585b4e3.png">

pimato commented:
looks like I can`t withdraw my OMN tokens with the max button: 
<img width="2053" alt="Bildschirmfoto 2021-04-21 um 11 31 18" src="https://user-images.githubusercontent.com/5337809/115531496-22a50900-a295-11eb-9ce3-108797cfc7d1.png">

pimato commented:
Instead of showing the transaction confirmed modal and user needs to get back to the `Your Connection - Modal` manually, we should transition from `Transaction modal - Pending Confirmation State` to the `Your Connection modal` directly with an updated OMN balance so the user sees the direct result of his action.

Mi-Lan committed:
Merge branch 'master' of https://github.com/protofire/omen-exchange into feature/1816-omn-bridge

Mi-Lan commented:
> After closing the confirmed transaction modal I went to the `Your Connection - Modal` and it seems like the balance is not getting updated after we did a successful deposit:
> <img alt="Bildschirmfoto 2021-04-21 um 11 11 22" width="440" src="https://user-images.githubusercontent.com/5337809/115528568-73ffc900-a292-11eb-8fd5-45f1a531e90d.png">
> 
> after refresh:
> <img alt="Bildschirmfoto 2021-04-21 um 11 12 30" width="444" src="https://user-images.githubusercontent.com/5337809/115528688-8f6ad400-a292-11eb-82ea-329e52f77309.png">

Hmm yeah seems that's the case for Dai also i will see to optimize deposit/withdraw function to encompass it...

Mi-Lan committed:
fixed set allowance not showing,incorrect withdraw max

Mi-Lan committed:
fixed margins

Mi-Lan committed:
added redirect into your connection modal after succesfull depositWithdraw

Mi-Lan committed:
Merge branch 'master' into feature/1816-omn-bridge

hexyls commented:
Hey man, I deployed a new [MultiClaim contract](https://etherscan.io/address/0xbbc9f0b5f1243946c42cdd6bc2267604b4f9a313#code) for you and you can just pass in an additional bridge address like you suggested.

I had a poke around the omnibridge docs just so I could get a better understanding. Those docs are so outdated lol anyway I eventually discovered that there is a [third bridge address](https://etherscan.io/address/0x4C36d2919e407f0Cc2Ee3c993ccF8ac26d9CE64e#readProxyContract) for the return trip from xdai->mainnet and this is where we send claim transactions to.

So bridge addresses are:

eth -> xdai bridge: 0x88ad09518695c6c3712AC10a214bE5109a655671
xdai -> eth bridge: 0xf6A78083ca3e2a662D6dd1703c939c8aCE2e268d
claim to wallet: 0x4C36d2919e407f0Cc2Ee3c993ccF8ac26d9CE64e

Mi-Lan commented:
> MultiClaim contract

@hexyls Awesome man! Thank you! But I will include multi claim in separate pr...

Mi-Lan committed:
fixed outdated margins

pimato reviewed (changes requested):
<img width="657" alt="Bildschirmfoto 2021-04-22 um 14 37 27" src="https://user-images.githubusercontent.com/5337809/115715404-5f95fc00-a378-11eb-8e65-7721b5217cea.png">
<img width="699" alt="Bildschirmfoto 2021-04-22 um 14 37 22" src="https://user-images.githubusercontent.com/5337809/115715406-602e9280-a378-11eb-892b-3a1d3206e79f.png">

Asset Description should be capitalized.

pimato commented:
I want to enable Dai on a new Account and it seems like I can`t load the asset icon: 
<img width="456" alt="Bildschirmfoto 2021-04-22 um 14 51 53" src="https://user-images.githubusercontent.com/5337809/115717477-82291480-a37a-11eb-8545-1d3d7e693e87.png">

Even though the link works just fine:
https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/assets/0x6B175474E89094C44Da98b954EedeAC495271d0F/logo.png

Mi-Lan committed:
fixed capital letter and added correct icons for enable

pimato committed:
Merge branch 'master' into feature/1816-omn-bridge

pimato committed:
Merge branch 'master' into feature/1816-omn-bridge

Mi-Lan committed:
Merge branch 'master' of https://github.com/protofire/omen-exchange into feature/1816-omn-bridge

Mi-Lan committed:
fixed set allowance not showing and number of optimisations

Mi-Lan committed:
Merge branch 'master' into feature/1816-omn-bridge

Mi-Lan committed:
Merge branch 'master' of https://github.com/protofire/omen-exchange into feature/1816-omn-bridge

Mi-Lan committed:
Merge branch 'feature/1816-omn-bridge' of https://github.com/protofire/omen-exchange into feature/1816-omn-bridge

Mi-Lan committed:
Merge branch 'master' into feature/1816-omn-bridge

pimato commented:
Remove decimals for any asset in the info box:
<img width="401" alt="Bildschirmfoto 2021-04-28 um 16 49 09" src="https://user-images.githubusercontent.com/5337809/116424321-b486b600-a841-11eb-95b1-33450d9feb55.png">

pimato commented:
Reset Inputfield on Asset switch:
![Bildschirmaufnahme 2021-04-28 um 16 57 11](https://user-images.githubusercontent.com/5337809/116425910-f6fcc280-a842-11eb-8cd2-d776254bf25f.gif)

Currently it anything the user has put in.

pimato commented:
We need to explain the user why the withdrawal of ERC-20 tokens does not work if the users omen account is not funded with Dai:
![Withdraw](https://user-images.githubusercontent.com/5337809/116430264-fb2adf00-a846-11eb-9e47-f698923ba191.png)

Note: 
- Replace min withdrawal amount message with the message above.

Figma:
[Design Specification](https://www.figma.com/file/R056tSYMjvfwaFidEBItYy/Widgets-V2?node-id=3832%3A241)

Mi-Lan committed:
added new message,updated input field and removed decimals

pimato reviewed (changes requested):
After I deposited/withdrew funds the appropriate balances have not been updated in the my connection modal.

Mi-Lan committed:
small optimizations to try to solve balance

Mi-Lan committed:
removed one unnceseary fetch

Mi-Lan committed:
Merge branch 'master' into feature/1816-omn-bridge

Mi-Lan commented:
@KadenZipfel @hexyls There is some problem with the fetching of balance for xOmen even after refreshing couple of times it doesn't update....Would be good if you guys can provide some feedback. I haven't been able to figure it out.
in useConnectedBalance I also tried to get balance with useCollateral but it was doing infinite loop for some reason i even left comments there...

hexyls committed:
fix infinite loop + add waitForBridgeMessageStatus

hexyls commented:
Hey @Mi-Lan, I took a look at this - the infinite loop in `useCollateralBalance` was caused by `collateral` in the dependency array. The `collateral` value comes from `getToken` which returns a new object with each render. I updated the dependency array in `useCollateralBalance` to use `collateral.symbol` instead of the whole object which sorted it out.

The other issue was token balances not updating, this turns out to be because the validators are really, really slow on the Omni bridge. Waiting 9 confirmations was not guaranteeing that tokens had been relayed, so when we refetched balances they hadn't actually changed yet. I added `waitForBridgeMessageStatus` that will wait until the Omni bridge validators have actually processed the deposit, so when we fetch balances after a tx we can be sure they are up to date.

Mi-Lan commented:
> Hey @Mi-Lan, I took a look at this - the infinite loop in `useCollateralBalance` was caused by `collateral` in the dependency array. The `collateral` value comes from `getToken` which returns a new object with each render. I updated the dependency array in `useCollateralBalance` to use `collateral.symbol` instead of the whole object which sorted it out.
> 
> The other issue was token balances not updating, this turns out to be because the validators are really, really slow on the Omni bridge. Waiting 9 confirmations was not guaranteeing that tokens had been relayed, so when we refetched balances they hadn't actually changed yet. I added `waitForBridgeMessageStatus` that will wait until the Omni bridge validators have actually processed the deposit, so when we fetch balances after a tx we can be sure they are up to date.

Thank you @hexyls 🤗

pimato commented on app/src/components/common/icons/IconOmen.tsx:
```diff
@@ -1,50 +1,44 @@
+}
+
+const SvgStyling = styled.svg<{ dropShadow?: boolean }>`
+  ${props => props.dropShadow && 'filter: drop-shadow(3px 3px 2px rgba(0, 0, 0, 0.7));'};
```
lets use: filter: drop-shadow(0px 2px 4px rgba(13, 71, 161, 0.25)

Mi-Lan committed:
changed drop shadow

Mi-Lan committed:
changed decimals to be 2 for omen balance

Mi-Lan committed:
fixed xOmen balance not updating after switching from xDai to Mainnet

Mi-Lan committed:
added withdraw fix

Mi-Lan committed:
Multi Claim  (#1852)

* made new claimable asset modal with expandable balance section

* logic for fetching claimable omen in progress

* finished fetching of unclaimed transactions and displaying of aggregated amounts

* added logic for displaying correct omen or dai text depeing on claim state

* optimized fetching of unclaimed transactions and cleaned up code

* started working on transaction modal

* added new multi claim contract implementation

* updated modal to not show dai icon on claim and figuring out whats wrong with dai batched

* found the wrong address and fixed it

* deleted some unusued stuff,renamed and optmized code for review

* waitForBridgeMessageStatus for deposits only

* reconfigured html to support hover for claimable assets and svg

* changed hover color for chevron down

* added cursor pointer to claimable assets

Co-authored-by: My Violet <milanrasovic.f@gmail.com>
Co-authored-by: hexyls <hexyls@protonmail.com>

hexyls reviewed (approved):
🙌 looks good!

pimato committed:
Merge branch 'master' into feature/1816-omn-bridge

KadenZipfel commented on app/src/services/xdai.ts:
```diff
@@ -135,15 +192,42 @@ class XdaiService {
+        const log = transaction.logs[transaction.logs.length - 1]
+        const messageId = log.topics[log.topics.length - 1]
+        // check if the message has been processed
+        const xdaiOmniAddress = '0x75Df5AF045d91108662D8080fD1FEFAd6aA0bb59'
```
This address should be a constant right?

Mi-Lan replied:
Dunno how that slipped! Good catch

kadenzipfel reviewed (changes requested):
Looking really good! Just some small suggestions here. I'm gonna do some proper testing in my next run through when I hopefully have more time

Mi-Lan commented on app/src/components/modal/modal_deposit_withdraw/index.tsx:
```diff
@@ -84,45 +121,119 @@ export const ModalDepositWithdraw = (props: Props) => {
+    !wallet ||
+    displayFundAmount.gt(wallet) ||
+    displayFundAmount.isZero() ||
+    !wallet ||
+    displayFundAmount.gt(wallet) ||
```
good catch!

Mi-Lan committed:
renamed old functions, delted unusued code and moved variable to constants

kazuolol committed:
Feature/1841 omn button (#1881)

* Add OMN Button in Navigation

* added OMN balance and icon to header

* added props to height/width of IconOmen svg

* changed from .toString to template literal omen balance

* use formatted balances for UI display

* removed decimals with slice

* removed drop shadow and resized icon

* fixed color and balance format

pimato reviewed (approved):
great work!

pimato committed:
Merge branch 'master' into feature/1816-omn-bridge

kadenzipfel reviewed (approved):
Haven't tested, but code looks 👌

Scheusal13 reviewed (commented):
0xD4EBBC1fE630e1aF1e2143d11b2C19c48d099098
